src/mixins/filing-template-mixin.ts

```diff
@@ -872,9 +872,22 @@ export default class FilingTemplateMixin extends DateMixin {
 
     // store People And Roles
     let orgPersons = filing.correction.parties || entitySnapshot?.orgPersons || []
-    // exclude Completing Party
-    // (it is managed separately and added to the filing in buildCorrectionFiling())
-    orgPersons = orgPersons.filter(op => !(op?.roles.some(role => role.roleType === RoleTypes.COMPLETING_PARTY)))
+
+    // filter in only the roles we currently support
+    orgPersons.forEach(op => {
+      op.roles = op.roles.filter(role => {
+        if (this.isBaseCorrectionFiling) return (role.roleType === RoleTypes.DIRECTOR)
+        if (this.isCoopCorrectionFiling) return (role.roleType === RoleTypes.DIRECTOR)
+        if (this.isFirmCorrectionFiling) {
+          return ([RoleTypes.PARTNER, RoleTypes.PROPRIETOR].includes(role.roleType))
+        }
+        return false // should never get here
```

Comment:
I did it this way to support upcoming changes (new party roles that we might need to correct, eg, officers, reviewers, significant individuals, maybe others).

Note that this `filter` function may return empty array (`[]`).

Reply:
I did not update Change or Conversion filings, but those are for firms which are not impacted by current activities (and have been in place for years), so those should be OK for now but may need an update if/when additional party roles are applied to them.